### ISSUE TITLE
twisted.positioning should be ported to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7987]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7987|
|Type|enhancement|
|Created|2015-08-05 06:53:33Z|
|Branch|https://github.com/twisted/twisted/tree/positioning-py3-7987|

<details><summary>Searchable metadata</summary>

```
trac-id__7987 7987
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_positioning_py3_7987 branches/positioning-py3-7987
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1438757613064556 1438757613064556
changetime__1439206884468964 1439206884468964
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [45395]) Branching to positioning-py3-7987.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Tests look green. Coverage looks good. I think I've made everything bytes that should be in the docs. One spurious pyflakes error (it's not a failure on Py3-ran Pyflakes).

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Thanks for working on this.

* Why not make FancyEqMixin inherit directly from `object` ?
* Please try to fix the pyflake error. You can use twistd.python.reflect.namedModule to avoid the pyflake errors. I know that this is a bit of extra work but I would like to have pyflakes errors clean and not get back to the mess we have with twistedchecker reported errors.

Please address my commends and merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [45437]) Merge positioning-py3-7987: Port twisted.positioning to Python 3

Author: hawkowl
Reviewer: adiroiban
Fixes: [#7987](https://github.com/twisted/twisted/issues/7987)
